src/Tizen.Network.Nsd/Tizen.Network.Nsd/DnssdService.cs

```diff
@@ -252,11 +307,8 @@ public void AddTXTRecord(string key, string value)
                 NsdErrorFactory.ThrowDnssdException(ret);
             }
 
-            byte[] txtValue;
-            ushort txtLength;
-            GetTxtRecord(out txtLength, out txtValue);
-
-            ret = Interop.Nsd.Dnssd.SetRecord(_serviceHandle, _dnsRecordtype, txtLength, txtValue);
+            byte[] txtValue = GetRawTxtRecord();
+            ret = Interop.Nsd.Dnssd.SetRecord(_serviceHandle, _dnsRecordtype, (ushort)txtValue.Length, txtValue);
```

Comment:
Could you expain why need to `SetRecord`?

Reply:
SetRecord used to set which type of record to add (please see the _dnsRecordtype param in setRecord)

Reply:
Why `Interop.Nsd.Dnssd.AddTxtRecord(_serviceHandle, key, length, byteValue)` does not work as you say? if  native `AddTxtRecord` is working correctly, it seems not need.

TizenFX is a c# port of Tizen Device API, So I think we should provide a same semantics on API, but your code does not.
AddTXTRecord on TizenFX, is AddTxtRecord + SetRecord on native api